bitcoin/src/address/error.rs

```diff
@@ -10,41 +10,25 @@ use crate::prelude::*;
 use crate::{base58, Network};
 
 /// Network error inside address.
+/// Address's network differs from required one.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum NetworkError {
-    /// Address's network differs from required one.
-    NetworkValidation {
-        /// Network that was required.
-        required: Network,
-        /// The address itself
-        address: Address<NetworkUnchecked>,
-    },
+pub struct NetworkValidationError {
+    /// Network that was required.
+    pub required: Network,
```

Comment:
Given our past discussions around `Network` we should probably make this private for now.

Reply:
We should probably have an accessor that returns only the prefix that failed to match. Or no accessor, for now, since it's easy to add one in a nonbreaking way.

Reply:
By the time `require_network` is called the input string is gone, so we don't have the prefix.

Reply:
We have the whole `Address` in the error so we can use it to compute `expected_prefix(&self) -> &'static str` and `provided_prefix(&self) -> &'static str` However we have to make error immutable in that case. (Which is probably a good idea anyway.)